### PR TITLE
Distinguish errors for non-existing LocalQueue and ClusterQueue in the workload-controller

### PR DESCRIPTION
**Pull request by tenzen-y**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:
In the https://github.com/kubernetes-sigs/kueue/pull/3605, we dropped the inactive LocalQueue and ClusterQueue reason.
But we still face errors in situations where those are inactive.

Additionally, we should not omit errors verifications in testing.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: pkg/controller/core/workload_controller.go [hunk 1]
```diff
@@ -607,9 +608,16 @@ func (r *WorkloadReconciler) Create(e event.CreateEvent) bool {
...
+		if err := r.queues.AddOrUpdateWorkload(wlCopy); err != nil {
+			var msg string
+			baseMsg := "for workload didn't exist or not active; ignored for now"
+			switch {
+			case errors.Is(err, queue.ErrClusterQueueDoesNotExist):
+				msg = fmt.Sprintf("ClusterQueue %s", baseMsg)
+			case errors.Is(err, queue.ErrQueueDoesNotExist):
+				msg = fmt.Sprintf("LocalQueue %s", baseMsg)
+			}
+			log.V(2).Info(msg)
```
Comment (mimowo) [inline pkg/controller/core/workload_controller.go:620]: Hm `ErrClusterQueueDoesNotExist` already contains the information this is about "clusterQueue":  https://github.com/kubernetes-sigs/kueue/blob/cc65f7e7701df31e77959ef664384616abd54f9e/pkg/queue/manager.go#L42-L43. So why to duplicate the information?

Also, this is a bulk of duplicated code, please commonize.

File: pkg/queue/manager_test.go [hunk 2]
```diff
@@ -533,7 +532,10 @@ func TestStatus(t *testing.T) {
 		}
 	}
 	for _, wl := range workloads {
-		_ = manager.AddOrUpdateWorkload(&wl)
+		// We ignore the ErrClusterQueueDoesNotExist since we never set up ClusterQueue in this test, and the error should be occurred.
+		if err := manager.AddOrUpdateWorkload(&wl); err != nil && !errors.Is(err, ErrClusterQueueDoesNotExist) {
+			t.Fatalf("Failed to add or update workloads: %v", err)
+		}
```
Comment (mimowo) [inline pkg/queue/manager_test.go:538]: I think this change is very good and super safe, if you extract to separate PR we can merge immediately

File: pkg/controller/core/workload_controller.go (hunk 1 above)
Comment (tenzen-y) [reply pkg/controller/core/workload_controller.go:620]: That makes sense. Here, key problem is dropped the "inactive" reason from logging message and mention "queue" instead of localQueue.

https://github.com/kubernetes-sigs/kueue/pull/3605/files#diff-60dd240c20adbd6a189d018d1c216c2d296730f446c341d8bf449fa6657964ffL733

So, I will remove these error-typed checkings and then add an "inactive" reason.

File: pkg/queue/manager_test.go (hunk 2 above)
Comment (tenzen-y) [reply pkg/queue/manager_test.go:538]: Opened: https://github.com/kubernetes-sigs/kueue/pull/3838

File: pkg/controller/core/workload_controller.go [hunk 3]
```diff
@@ -607,9 +607,8 @@ func (r *WorkloadReconciler) Create(e event.CreateEvent) bool {
 	workload.AdjustResources(ctx, r.client, wlCopy)
 
 	if !workload.HasQuotaReservation(wl) {
-		err := r.queues.AddOrUpdateWorkload(wlCopy)
-		if err != nil {
-			log.V(2).Info(fmt.Sprintf("%s; ignored for now", err))
+		if err := r.queues.AddOrUpdateWorkload(wlCopy); err != nil {
+			log.V(2).Info("ignored an error for now", "error", err)
```
Comment (tenzen-y) [inline pkg/controller/core/workload_controller.go:611]: It would be better to use structured logging instead of wrapping values inside the message.

File: pkg/controller/core/workload_controller.go (hunk 1 above)
Comment (tenzen-y) [reply pkg/controller/core/workload_controller.go]: Done.
Please let me know if you have more recommendations.
Thanks.

File: pkg/queue/manager.go [hunk 4]
```diff
@@ -39,8 +39,8 @@ import (
 )
 
 var (
-	ErrQueueDoesNotExist         = errors.New("queue doesn't exist")
-	ErrClusterQueueDoesNotExist  = errors.New("clusterQueue doesn't exist")
+	ErrQueueDoesNotExist         = errors.New("localQueue doesn't exist or inactive")
```
Comment (mbobrovskyi) [inline pkg/queue/manager.go:42]:
```suggestion
	ErrLocalQueueDoesNotExistOrInactive         = errors.New("localQueue doesn't exist or inactive")
```

File: pkg/queue/manager.go [hunk 5]
```diff
@@ -39,8 +39,8 @@ import (
...
+	ErrClusterQueueDoesNotExist  = errors.New("clusterQueue doesn't exist or inactive")
```
Comment (mbobrovskyi) [inline pkg/queue/manager.go:43]:
```suggestion
	ErrClusterQueueDoesNotExistOrInactive  = errors.New("clusterQueue doesn't exist or inactive")
```

File: pkg/queue/manager.go (hunk 4 above)
Comment (tenzen-y) [reply pkg/queue/manager.go:42]: SGTM, thanks.

Comment (tenzen-y) [reply pkg/queue/manager.go]: Done.

File: pkg/queue/manager.go (hunk 5 above)
Comment (tenzen-y) [reply pkg/queue/manager.go]: Done.

**Comment by tenzen-y** [comment]
@mbobrovskyi  I addressed all your comments.

**Comment by mbobrovskyi** [comment]
LGTM. Thank you so much!

**Comment by mbobrovskyi** [comment]
Left lgtm for @mimowo

File: pkg/controller/core/workload_controller.go (hunk 1 above)
Comment (mimowo) [reply pkg/controller/core/workload_controller.go]: I see, but do you have a code reference (could be a unit test) indicating that the error is returned for inactive queues (this is something I'm not sure about).

Comment (mbobrovskyi) [reply pkg/controller/core/workload_controller.go]: We add LocalQueue to the queue manager only if it is active.

https://github.com/kubernetes-sigs/kueue/blob/5ad7e23fdeed50d90d90629c28623f592063ac13/pkg/controller/core/localqueue_controller.go#L141-L146

Comment (mimowo) [reply pkg/controller/core/workload_controller.go]: I see, so the message "does not exist or inactive" is accurate.  Do we have any unit tests for this scenario?

How about ClusterQueue? In case it is inactive we still expect it in cache here: https://github.com/kubernetes-sigs/kueue/blob/5ad7e23fdeed50d90d90629c28623f592063ac13/pkg/cache/cache.go#L333-L340

Comment (mbobrovskyi) [reply pkg/controller/core/workload_controller.go]: Agree. For CQ we need to revert.

Comment (tenzen-y) [reply pkg/controller/core/workload_controller.go]: As I discussed with @mimowo, for LocalQueue, the "inactive" reason is valid. But for ClusterQueue, we always insert that to queueManager, and then we identify the active CQs by `Active` function: https://github.com/kubernetes-sigs/kueue/blob/5ad7e23fdeed50d90d90629c28623f592063ac13/pkg/queue/cluster_queue.go#L58

So, we decided to remove the "inactive" reason only from ClusterQueue.

Comment (tenzen-y) [reply pkg/controller/core/workload_controller.go]: Done

**Comment by mimowo** [comment]
/lgtm
/approve
please also cherry-pick this one as a follow up to the previous fix
